src/fides/api/alembic/migrations/versions/f17f92237383_disable_inactive_notices.py

```diff
@@ -0,0 +1,151 @@
+"""disable_inactive_notices
+
+Revision ID: f17f92237383
+Revises: 671358247251
+Create Date: 2023-09-14 13:35:38.474968
+
+"""
+import uuid
+from typing import Set
+
+from alembic import op
+from sqlalchemy import text
+from sqlalchemy.engine import Connection, ResultProxy
+from sqlalchemy.sql.elements import TextClause
+
+
+# revision identifiers, used by Alembic.
+from fides.api.models.sql_models import DataUse
+
+revision = "f17f92237383"
+down_revision = "671358247251"
+branch_labels = None
+depends_on = None
+
+
+def get_expanded_system_data_uses(bind: Connection) -> Set:
+    """Get data uses across all systems and those data uses' parents
+
+    Same results as System.get_data_uses(System.all(db), include_parents=True), but
+    we don't want to use the System Sqlalchemy model within a migration
+    """
+    system_data_uses: ResultProxy = bind.execute(
+        text("SELECT data_use from privacydeclaration;")
+    )
+
+    expanded_data_uses = set()
+    for row in system_data_uses:
+        data_use: str = row["data_use"]
+        # Expand system data uses so we also include their parents
+        expanded_data_uses.update(DataUse.get_parent_uses_from_key(data_use))
+    return expanded_data_uses
+
+
+def disable_privacy_notice_and_bump_version(
+    bind: Connection, privacy_notice_id: str, current_version: int
+) -> None:
+    """Disable the PrivacyNotice and bump its version"""
+    disable_privacy_notice_query: TextClause = text(
+        "UPDATE privacynotice SET disabled=true, version=:new_version WHERE id = :privacy_notice_id;"
+    )
+    bind.execute(
+        disable_privacy_notice_query,
+        {
+            "privacy_notice_id": privacy_notice_id,
+            "new_version": int(current_version) + 1,
+        },
+    )
+
+
+def create_corresponding_historical_record(
```

Comment:
Rather than just disabling the current record , I am disabling, bumping the version, and creating a historical record, to keep a good history for consent reporting.  (This is what would happen if we disabled in the UI, too)

It is definitely possible preferences have been saved against enabled privacynoticehistory records that I am disabling.  If I just disable here, without any audit trail, consent reporting will look like we made it possible for a user to save a preference against a disabled notice.

Reply:
wow, good thinking!!